python/pyiceberg/expressions/literals.py

```diff
@@ -108,7 +110,7 @@ def __ge__(self, other):
 
 
 @singledispatch
-def literal(value) -> Literal:
+def literal(value: Any) -> Literal:
```

Comment:
Shouldn't this be `value: T` and `Literal[T]`?

Reply:
Yes, but this opens up many more type errors which I'd like to do in a separate PR.